### DISCUSSION TITLE
INSERT INTO 数据丢失

### DISCUSSION THREAD

**Discussion by HigherLove520** [apache/doris#34222] [A - General / Q&A]
使用JDBC的方式去Doris执行INSERT INTO table SELECT * FROM other_table Conditions; 存在数据丢失情况，other_table中的所有列每一行数据长度和类型都是一样的，应该不存在有长度和类型不匹配的问题，开启了严格模式也不起作用。
![image](https://github.com/apache/doris/assets/99522535/0fc06fa5-e552-4301-a8fc-fd55d291ea94)

**Comment by yagagagaga** [comment]
Can you show me the table's ddl?

**Reply by HigherLove520** [reply]
ok, thank you for your replay. here is my table's ddl 
![image](https://github.com/apache/doris/assets/99522535/bda8f760-7a6f-4714-b9c7-81a882b156bf)

I have a view table and an entity table, with the view having the same number of columns and names as the entity table. When I use INSERT INTO to import data from the view table to the entity table, data loss occurs. Strangely, the data that is lost and the data that is not lost are exactly the same in length and type, as they are simulated fake data by the program.

**Reply by yagagagaga** [reply]
Is there duplicate data in the `METER_NO` column?

**Reply by HigherLove520** [reply]
Thank you for your reply. There are no duplicates of METER_NO. I am implementing a feature to move data from a complex view to an entity table. When inserting data from the view into the entity table, data loss always occurs. METER_NO is unique in both the view and the entity table. When executing the INSERT via JDBC, the executeUpdate method returns successfully for 50 records, but only 30 records appear in the entity table.

**Reply by HigherLove520** [reply]
![image](https://github.com/apache/doris/assets/99522535/30018804-975e-4c66-8657-6a8cbc7a9dc9)
The same import statement is executed twice in the unique model at the same time. The ReturnRows in the log are different. The previous statement indicates that I lost 10 pieces of data during the import process, while the subsequent execution shows that I did not lose any data in this Insert INTO import

**Reply by yagagagaga** [reply]
It is very strange, there must be a mistake somewhere. Maybe you can join the community and we have deeper communication. https://github.com/apache/doris/discussions/16887

**Comment by gongxk27** [comment]
解决了吗，我在2.17版本也遇到了，表里本来有1200多行不重复的数据，插入几十条后只剩1000条了，这也太可怕了吧。
